public/js/post/post.js

```diff
@@ -63,47 +63,59 @@ var MultipleCourseCode = React.createClass({
             infoOpened: false
         }
     },
+
+    componentDidMount: function() {
+        for (i = 0; i < this.props.data.courses.length; i++) {
+            this.setState({completedTextBoxes: this.state.completedTextBoxes += 1});
```

Comment:
@arkon I actually tried that. For some really strange reason, when I try this, `this.state.completedTextboxes` is still 0 right after, so the `this.checkIfCompleted()` call right after it returns false even when it's really true. However when I check the state in Chrome React tools after it's all been rendered, `this.state.completedTextboxes` has properly changed. I can't figure out why that is. Somehow the for loop causes the state to update automatically.